Remove deprecated code for version 3005 "Phosphorus"

bdrung opened the pull request:
As of commit 52228209f317760bcf296b25f8c91155227219ec, the current master branch contains the upcoming 3005 "Phosphorus" code. Several test cases throw a `RuntimeError` about deprecation warnings for version 3005.

Therefore remove the warnings (as advised by the `RuntimeError`) which includes removing:

* class `MultiprocessingProcess`
* class `SignalHandlingMultiprocessingProcess`
* legacy `generate_sls_context`
* `nxos.add_config` (use `nxos.config` instead)
* `nxos.cmd COMMAND` (use `nxos.COMMAND` instead)
* `nxos.show` (use `nxos.sendline` instead)
* `nxos.system_info` (use `nxos.grains` instead)

bdrung committed:
nxos module: Replace deprecated nxos.cmd

Replace deprecated `nxos.cmd COMMAND` by `nxos.COMMAND`.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
nxos module: Replace deprecated nxos.add_config

Replace deprecated `nxos.add_config` by `nxos.config`.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
nxos module: Replace deprecated nxos.show

Replace deprecated `nxos.show` by `nxos.sendline`.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

Commit:
Remove enable_slsvars_fixes feature flag and enable behavior by default

bdrung commented:
Rebased and resolve conflict.

Ch3LL reviewed (commented):
Thanks for all of this. Can you just add a changelog for the removals:

1. https://github.com/saltstack/salt/pull/61718/files#diff-9417515db532635f683c548ecd2f6edf210c983b77b4602d522476aedd9ebc38  for slsvars one
[EDIT]
2. A `.removed` changed for removing MultiprocessingProcess and SignalHandlingMultiprocessingProcess

This will also close out https://github.com/saltstack/salt/issues/61706  thanks!

bdrung committed:
Remove deprecated code for version 3005 "Phosphorus"

As of commit 52228209f317760bcf296b25f8c91155227219ec, the current
master branch contains the upcoming 3005 "Phosphorus" code. Several test
cases throw a `RuntimeError` about deprecation warnings for version
3005.

Therefore remove the warnings (as advised by the `RuntimeError`) which
includes removing:

* class `MultiprocessingProcess`
* class `SignalHandlingMultiprocessingProcess`

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung commented:
I cherry-picked your commit for removing `_generate_sls_context_legacy` and added `changelog/61573.removed`.

Commit:
Merge branch 'master' into remove-deprecated-code
